### PR TITLE
[indexer][graphql] Pruner prunes only epoch-partitioned tables

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 

Currently, the pruner assumes that all partitioned tables are partitioned on epoch, which is an issue since `objects_version` is not partitioned by epoch. Modify the pruner so that it will filter out non-epoch-partitioned tables, and otherwise do the same thing.

Change `EPOCHS_TO_KEEP` from an env variable to a config, so we can pass in test values through the transactional test runner, and a prune.move test to validate that we prune epoch data without any trouble.

## Test plan 

prune.move

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-indexer/src/handlers/pruner.rs [hunk 1]
```diff
@@ -53,18 +54,28 @@ impl Pruner {
...
+            let table_partitions: HashMap<_, _> = self
+                .partition_manager
+                .get_table_partitions()?
+                .into_iter()
+                .filter(|(table_name, _)| {
+                    self.partition_manager
+                        .get_strategy(table_name)
+                        .is_epoch_partitioned()
+                })
+                .collect();
```
Comment (gegaowp) [inline crates/sui-indexer/src/handlers/pruner.rs:67]: 👍

File: crates/sui-indexer/src/indexer.rs [hunk 2]
```diff
@@ -101,9 +103,6 @@ impl Indexer {
         )
         .await?;
 
-        let epochs_to_keep = std::env::var("EPOCHS_TO_KEEP")
```
Comment (gegaowp) [inline crates/sui-indexer/src/indexer.rs:104]: we will still need to keep the env reading, maybe in `start_writer`? b/c this is enabled in production like https://github.com/MystenLabs/sui-operations/blob/main/pulumi/gcp/sui-indexer/Pulumi.mainnet.yaml#L42
it's doable but more changes are needed to entirely get rid of EPOCHS_TO_KEEP as an env var, maybe making it part of indexer config.

Comment (wlmyng) [reply crates/sui-indexer/src/indexer.rs:104]: ah oops, that's a good call

Comment (wlmyng) [reply crates/sui-indexer/src/indexer.rs:104]: added it back for now

**Review by gegaowp — APPROVED** [review]

**Pull request merged**